vck/src/commonMain/kotlin/at/asitplus/wallet/lib/agent/KeyMaterial.kt

```diff
@@ -2,27 +2,34 @@ package at.asitplus.wallet.lib.agent
 
 import at.asitplus.signum.indispensable.Digest
 import at.asitplus.signum.indispensable.ECCurve
+import at.asitplus.signum.indispensable.HMAC
 import at.asitplus.signum.indispensable.cosef.io.Base16Strict
 import at.asitplus.signum.indispensable.josef.JsonWebKey
 import at.asitplus.signum.indispensable.josef.toJsonWebKey
 import at.asitplus.signum.indispensable.pki.X509Certificate
 import at.asitplus.signum.indispensable.pki.X509CertificateExtension
+import at.asitplus.signum.indispensable.symmetric.SealedBox
+import at.asitplus.signum.indispensable.symmetric.SymmetricKey
 import at.asitplus.signum.indispensable.toX509SignatureAlgorithm
 import at.asitplus.signum.supreme.asKmmResult
+import at.asitplus.signum.supreme.mac.mac
 import at.asitplus.signum.supreme.sign.EphemeralKey
 import at.asitplus.signum.supreme.sign.Signer
+import at.asitplus.signum.supreme.symmetric.encrypt
 import io.github.aakira.napier.Napier
 import io.matthewnelson.encoding.core.Encoder.Companion.encodeToString
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 import kotlin.random.Random
 
+interface KeyMaterial {
+    val identifier: String
+}
+
 /**
  * Abstracts the management of key material away from cryptographic functions.
  */
-interface KeyMaterial : Signer {
-    val identifier: String
-
+interface SignKeyMaterial : KeyMaterial, Signer {
```

Comment:
I see the issue with `KeyMaterial`, but I'm not happy renaming it, as it involves rather large changes.

Would it work if we keep the name `KeyMaterial`, but introduce a new one named `SymmetricKeyMaterial`? Because I think current use of `KeyMaterial` always require it to be asymmetric, and the symmetric key should be a second constructor parameter if needed.